spec/services/openid_connect_redirector_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'rails_helper'
+
+RSpec.describe OpenidConnectRedirector do
+  include Rails.application.routes.url_helpers
+
+  let(:redirect_uri) { 'http://localhost:7654/' }
+  let(:state) { SecureRandom.hex }
+  let(:service_provider) { ServiceProvider.from_issuer('urn:gov:gsa:openidconnect:sp:server') }
+  let(:errors) { ActiveModel::Errors.new(nil) }
+
+  describe '.from_request_url' do
+    it 'builds a redirector from an OpenID request_url' do
+      request_url = openid_connect_authorize_url(
+        client_id: service_provider.issuer,
+        redirect_uri: redirect_uri,
+        state: state
+      )
+
+      result = OpenidConnectRedirector.from_request_url(request_url)
+
+      expect(result).to be_a(OpenidConnectRedirector)
+      expect(result.send(:redirect_uri)).to eq(redirect_uri)
```

Comment:
Ok thanks for talking through it with me! I am going to leave as-is because I feel better having a more precise test that uses some private methods rather than a loose test that might let a bug in a refactor slip through later,